tests/baselines/reference/tsbuild/configFileErrors/reports-syntax-errors-in-config-file-discrepancies.js

```diff
@@ -1,85 +1,4 @@
 0:: reports syntax errors after change to config file
 During incremental build, tsbuildinfo is not emitted, so declaration option is not present
 Clean build has declaration option in tsbuildinfo
-TsBuild info text without affectedFilesPendingEmit:: /src/tsconfig.tsbuildinfo.readable.baseline.txt::
-CleanBuild:
-{
-  "program": {
-    "fileInfos": {
-      "../lib/lib.d.ts": {
-        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
-        "affectsGlobalScope": true,
-        "impliedFormat": "commonjs"
-      },
-      "./a.ts": {
-        "version": "4646078106-export function foo() { }",
-        "impliedFormat": "commonjs"
-      },
-      "./b.ts": {
-        "version": "1045484683-export function bar() { }",
-        "impliedFormat": "commonjs"
-      }
-    },
-    "root": [
-      [
-        2,
-        "./a.ts"
-      ],
-      [
-        3,
-        "./b.ts"
-      ]
-    ],
-    "options": {
-      "composite": true,
-      "declaration": true
-    },
-    "referencedMap": {},
-    "changeFileSet": [
-      "../lib/lib.d.ts",
-      "./a.ts",
-      "./b.ts"
-    ]
-  },
-  "version": "FakeTSVersion"
-}
-IncrementalBuild:
-{
-  "program": {
-    "fileInfos": {
-      "../lib/lib.d.ts": {
-        "version": "3858781397-/// <reference no-default-lib=\"true\"/>\ninterface Boolean {}\ninterface Function {}\ninterface CallableFunction {}\ninterface NewableFunction {}\ninterface IArguments {}\ninterface Number { toExponential: any; }\ninterface Object {}\ninterface RegExp {}\ninterface String { charAt: any; }\ninterface Array<T> { length: number; [n: number]: T; }\ninterface ReadonlyArray<T> {}\ndeclare const console: { log(msg: any): void; };",
-        "affectsGlobalScope": true,
-        "impliedFormat": "commonjs"
-      },
-      "./a.ts": {
-        "version": "4646078106-export function foo() { }",
-        "impliedFormat": "commonjs"
-      },
-      "./b.ts": {
-        "version": "1045484683-export function bar() { }",
-        "impliedFormat": "commonjs"
-      }
-    },
-    "root": [
-      [
-        2,
-        "./a.ts"
-      ],
-      [
-        3,
-        "./b.ts"
-      ]
-    ],
-    "options": {
-      "composite": true
-    },
-    "referencedMap": {},
-    "changeFileSet": [
-      "../lib/lib.d.ts",
-      "./a.ts",
-      "./b.ts"
-    ]
-  },
-  "version": "FakeTSVersion"
-}
\ No newline at end of file
+*** Supplied discrepancy explanation but didnt file any difference
```

Comment:
This seems like it needs fixing?

Reply:
Ah, there are other files with this (and other threads), forgot.

Reply:
I've removed the small builder change in this PR - we can look into it separately. The issue there isn't related to the new  `noCheck` flag, but rather applies to any semantic-diagnostic-affecting option you build-via-api with a different setting than you build-via-cli with, which, as we discussed, needs some more extensive builder changes to properly support (and to support passing builder-wide semantic settings via CLI generally).